azurerm/internal/services/securitycenter/resource_arm_security_center_setting.go

```diff
@@ -0,0 +1,112 @@
+package securitycenter
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/securitycenter/azuresdkhacks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/securitycenter/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func resourceArmSecurityCenterSetting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSecurityCenterSettingUpdate,
+		Read:   resourceArmSecurityCenterSettingRead,
+		Update: resourceArmSecurityCenterSettingUpdate,
+		Delete: resourceArmSecurityCenterSettingDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"setting_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"MCAS",
+					"WDATP",
+				}, false),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceArmSecurityCenterSettingUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.SettingClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	settingName := d.Get("setting_name").(string)
+	enabled := d.Get("enabled").(bool)
+	setting := security.DataExportSettings{
+		DataExportSettingProperties: &security.DataExportSettingProperties{
+			Enabled: &enabled,
+		},
+		Kind: security.KindDataExportSettings,
+	}
+
+	if _, err := client.Update(ctx, settingName, setting); err != nil {
+		return fmt.Errorf("Error creating/updating Security Center pricing: %+v", err)
+	}
+	// TODO: switch to back when Swagger/API bug has been fixed:
+	// https://github.com/Azure/azure-sdk-for-go/issues/12687
+	resp, err := azuresdkhacks.GetSecurityCenterSetting(client, ctx, settingName)
+	if err != nil {
+		return fmt.Errorf("Error reading Security Center setting: %+v", err)
+	}
+
+	d.SetId(*resp.ID)
```

Comment:
We should probably have a nil check here for safety

```suggestion
	if resp.ID == nil || *resp.ID == "" {
		return fmt.Errorf("nil/empty ID returned for Security Center setting %q", settingName)
	}
	d.SetId(*resp.ID)
```